pkg/gofr/datasource/solr/solr.go

```diff
@@ -0,0 +1,215 @@
+package solr
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+)
+
+type Config struct {
+	Host string
+	Port string
+}
+
+type Client struct {
+	url string
+
+	logger  Logger
+	metrics Metrics
+}
+
+// New initializes Solr driver with the provided configuration.
+// The Connect method must be called to establish a connection to Solr.
+// Usage:
+// client := New(config)
+// client.UseLogger(loggerInstance)
+// client.UseMetrics(metricsInstance)
+// client.Connect()
+func New(conf Config) *Client {
+	s := &Client{}
+	s.url = "http://" + conf.Host + ":" + conf.Port + "/solr/"
+
+	return s
+}
+
+// UseLogger sets the logger for the Cassandra client which asserts the Logger interface.
+func (c *Client) UseLogger(logger any) {
+	if l, ok := logger.(Logger); ok {
+		c.logger = l
+	}
+}
+
+// UseMetrics sets the metrics for the Cassandra client which asserts the Metrics interface.
+func (c *Client) UseMetrics(metrics any) {
+	if m, ok := metrics.(Metrics); ok {
+		c.metrics = m
+	}
+}
+
+// Connect establishes a connection to Solr and registers metrics using the provided configuration when the client was Created.
+func (c *Client) Connect() {
+	c.logger.Infof("connecting to Solr at %v", c.url)
+
+	solrBuckets := []float64{.05, .075, .1, .125, .15, .2, .3, .5, .75, 1, 2, 3, 4, 5, 7.5, 10}
+	c.metrics.NewHistogram("app_solr_stats", "Response time of Solr operations in milliseconds.", solrBuckets...)
+
+	return
+}
+
+func (c *Client) HealthCheck(ctx context.Context) (any, error) {
+	return nil, nil
+}
+
+// Search searches documents in the given collections based on the parameters specified.
+// This can be used for making any queries to SOLR
+func (c *Client) Search(ctx context.Context, collection string, params map[string]any) (any, error) {
+	url := c.url + collection + "/select"
```

Comment:
https://github.com/gofr-dev/gofr/pull/985#discussion_r1760613131


```suggestion
	url := c.url + "/" + collection + "/select"
```